analytics/analytics.go

```diff
@@ -0,0 +1,102 @@
+package analytics
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/alecthomas/kong"
+	"github.com/denisbrodbeck/machineid"
+	"github.com/tcnksm/go-gitconfig"
+	"net/http"
+	"strings"
+	"time"
+)
+
+var (
+	// HeapBaseURI is the location of Heap API endpoint
+	HeapBaseURI = "https://heapanalytics.com"
+	// HeapAppID identifies what Heap App events are recorded against
+	HeapAppID = "4248790180"
+)
+
+// Private type for submitting server side events to Heap
+// https://developers.heap.io/reference#track-1
+type heapTrack struct {
+	AppID  string           `json:"app_id"`
+	Events []heapTrackEvent `json:"events"`
+}
+
+// Private type for submitting server side events to Heap
+// https://developers.heap.io/reference#track-1
+type heapTrackEvent struct {
+	Identity   string            `json:"identity"`
+	Timestamp  time.Time         `json:"timestamp"`
+	Event      string            `json:"event"`
+	Properties map[string]string `json:"properties"`
+}
+
+// Event records an interaction with the Section CLI
+type Event struct {
+	Name       string
+	Properties map[string]string
+}
+
+// Identity tries to determine the identity of:
+//
+// - the person running the cli
+// - the machine the cli is being run on
+func identity() (id string) {
+	id, err := gitconfig.Email()
```

Comment:
This should be prohibited without the explicit consent of the user. It is PII. It is completely illegal in Europe.

Reply:
Not to mention that some users might oppose any attempt to being tracked

Reply:
💯

There's a missing part to this PR – an explicit opt-in. 

I'm thinking I might just simplify this for now and only use the anonymized machine id.

Reply:
Fixed from 40d8ac30393e5f74efdece620d0e8e0af78469d2 onwards.